packages/expo-background-task/android/src/main/java/expo/modules/backgroundtask/BackgroundTaskScheduler.kt

```diff
@@ -0,0 +1,214 @@
+package expo.modules.backgroundtask
+
+import android.content.Context
+import android.os.Build
+import android.util.Log
+import androidx.work.Constraints
+import androidx.work.Data
+import androidx.work.ExistingPeriodicWorkPolicy
+import androidx.work.NetworkType
+import androidx.work.Operation
+import androidx.work.PeriodicWorkRequestBuilder
+import androidx.work.WorkInfo
+import androidx.work.WorkManager
+import com.google.common.util.concurrent.ListenableFuture
+import expo.modules.interfaces.taskManager.TaskServiceProviderHelper
+import kotlinx.coroutines.CancellationException
+import kotlinx.coroutines.CompletableDeferred
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.awaitAll
+import kotlinx.coroutines.withContext
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+class BackgroundTaskScheduler {
+  companion object {
```

Comment:
```suggestion
object BackgroundTaskScheduler {
```

if the whole class is in a companion object, we can just use `object`